### DISCUSSION TITLE
Compare not working

### DISCUSSION THREAD

**Discussion by gaborrosta** [community/community#50051] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I tried to compare changes between different tags on a repository I own, and it says that there is nothing to compare. However, it works fine when I try ```git diff v1.0.5...v1.0.6``` locally. And I tried to check the comparing page with a repository I have access to (and I do not own that), and a totally different repository and comparing works there fine. Furthermore, it works fine for another repository I own. Can someone help me with this issue?

**Comment by Anmol-Baranwal** [comment, marked as answer]
It's possible that there is an issue with the tags or the repository configuration that is causing the problem. Here are a few steps you can take to try and resolve the issue:

1. Verify that the tags exist: Double-check that the tags you are trying to compare actually exist in the repository by running the git tag command. Make sure the tag names are spelled correctly.

2. Check the tag format: Make sure that the tags you are comparing follow the correct format. Git uses a specific format for tags, which is vX.Y.Z, where X, Y, and Z are version numbers.

3. Check the repository configuration: Ensure that the repository configuration is correct, including the default branch, repository permissions, and access controls.

4. Try comparing branches instead of tags: If comparing tags still doesn't work, try comparing the branches that the tags are associated with. For example, you can compare the master branch with the v1.0.5 and v1.0.6 tags using the command git diff v1.0.5...master and git diff v1.0.6...master.

If none of these steps resolve the issue, you may want to reach out to GitHub support for further assistance. They can help you troubleshoot the problem and identify any potential issues with your repository configuration.

**Reply by gaborrosta** [reply]
Thank you for your reply. I've checked everything, and it is still not working. I will write to GitHub support.

**Reply by pacharrin** [reply]
i can confirm that is not working for me either, it was working fine a few hours ago.

particularly it doesn't work when you try to compare between two shas, if you do 1 sha and a branch name it works

here are some examples using the `github/docs` repo

[this wont work](https://github.com/github/docs/compare/0cd45208f79effa7f3b987f0b408134cc10f6ead..604f75d5849a6c8dc2aaa0c699c9c1232aed72d8) 

[this will](https://github.com/github/docs/compare/0cd45208f79effa7f3b987f0b408134cc10f6ead..main)

**Comment by briaguya0** [comment]
I encountered the same issue, earlier this morning tag compare was working, and now the exact same link for that tag compare is not

| compare type | working? |
| - | - |
| branch -> tag | :heavy_check_mark: |
| tag -> tag | :x: |
| commit sha -> commit sha | :x: |

**Reply by briaguya0** [reply]
to add context, nothing has been pushed/merged since the last time it was working

**Comment by pjcpjc** [comment]
I'm seeing this problem as well. I rely heavily on this feature and just now tested it quite carefully. Definitely not working like it used to. Hopefully @gaborrosta 's support ticket will create a resolution. 

Bear in mind, compare used to work for SHAs as well as tags, and it never required tags to start with a "v".

**Reply by gaborrosta** [reply]
I will update the discussion if I hear back from support.

**Comment by nick-f** [comment]
I also wrote to GitHub support about this and they just replied saying a fix is in place now.

Links like @pacharrin's [this wont work](https://github.com/github/docs/compare/0cd45208f79effa7f3b987f0b408134cc10f6ead..604f75d5849a6c8dc2aaa0c699c9c1232aed72d8) now work again.

🎉

**Comment by pjcpjc** [comment]
All my prior problems now appear resolved. Thanks for the help everyone, esp @gaborrosta

**Comment by gaborrosta** [comment]
As you are already aware, it is working again. 🎉

**Comment by mahdi-Maersk** [comment]
I have the same issue since early this week. Sometimes there is not any diff showing in GitHub UI nor GitHub desktop. But the `diff branchA branchB` still works correctly.
